armi/utils/gridEditor.py

```diff
@@ -0,0 +1,1783 @@
+        elif isinstance(assemDesign, tuple):
+            self._setActiveAssemID(self.pathControl.activateButton.GetId())
+        elif assemDesign is None:
+            self._setActiveAssemID(None)
+
+
+class GridGui(wx.ScrolledWindow):
+    """
+    Visual editor for grid blueprints.
+
+    This is the actual viewer that displays the grid and grid blueprints contents, and
+    responds to mouse events. Under the hood, it uses a wx.PseudoDC to handle the
+    drawing, which provides the following benefits over a regular DC:
+     - Drawn objects can be associated with an ID, allowing parts of the drawing to be
+       modified or cleared without having to re-draw everything.
+     - The IDs associated with the objects can be used to distinguish what was clicked
+       on in a mouse event (though the support for this isn't super great, so we do have
+       to do some of our own object disambiguation).
+
+    The ``drawGrid()`` method is used to re-draw the entire geometry, whereas the
+    ``applyAssem()`` method may be used to update a single assembly.
+    """
+
+    class Mode(enum.IntEnum):
+        """
+        Enumeration for what type of objects are currently being manipulated.
+
+        This can either be SPECIFIER, for laying out the initial core layout, or PATH
+        for manipulating fuel shuffling paths.
+        """
+
+        # We use these values to map between selections in GUI elements, so don't go
+        # changing them willy-nilly
+        SPECIFIER = 0
+        POSITION_IJ = 1
+        POSITION_RINGPOS = 2
+        PATH = 3
+
+        @property
+        def label(self):
+            if self == self.SPECIFIER:
+                return "Specifier"
+            elif self == self.PATH:
+                return "Shuffle Path"
+            elif self == self.POSITION_IJ:
+                return "(i, j)"
+            else:
+                return "(Ring, Position)"
+
+        @property
+        def isPosition(self):
+            return self == self.POSITION_IJ or self == self.POSITION_RINGPOS
+
+    def __init__(self, parent, bp=None, defaultGeom=geometry.CARTESIAN):
+        """
+        Create a new GridGui.
+
+        Parameters
+        ----------
+        parent : wx.Window
+            The parent control
+
+        bp : Optional set of grid blueprints
+            This should be the ``gridDesigns`` section of a root Blueprints object. If
+            not provided, a dictionary will be created with an empty "core" grid
+            blueprint.
+        """
+        wx.ScrolledWindow.__init__(
+            self, parent, wx.ID_ANY, (0, 0), size=(250, 150), style=wx.BORDER_DEFAULT
+        )
+
+        self.parent = parent
+
+        if bp is None:
+            bp = {
+                "core": GridBlueprint(
+                    name="core", gridContents=dict(), geom=defaultGeom
+                )
+            }
+
+        self.bp = bp
+        self.coreBp = bp["core"]
```

Comment:
I kind of want to leave this as it is, since it is what it is. If we are hard-coding it to the core blueprint, then that's the name for it. We do get some abstraction from the `activeBlueprints` which does insulate the rest of the world from the `coreBp` member. Come extension time, we would add a new control to select the active grid/blueprints, figure out how to populate the object palette, etc. but most operations will still be operating through `activeBlueprints`.